### ISSUE TITLE
Legend doesn't scale on large exports

### ISSUE THREAD

**Issue by ntjess** [pyqtgraph/pyqtgraph#1552]
```python
import pyqtgraph as pg
import numpy as np
app = pg.mkQApp()
pw = pg.plot()

ii = pg.ImageItem()
ii.setBorder(pg.mkPen('r', width=5))
ii.setImage(np.random.random((6000,6000,3)))
pw.addItem(ii)

spi = pg.ScatterPlotItem()
spi.setData([0, 2000, 5000], [0, 2000, 5000], symbol='s', width=5, pxMode=True, size=10)
pw.addItem(spi)

leg = pg.LegendItem()
leg.setParentItem(pw.plotItem)
leg.addItem(spi, name='test')
leg.setPen('k')
leg.setBrush('k')

# Open "Export" menu, Image Export, set resolution to 6000x4500
# Legend is miniscule!
```
### Expected behavior
Legend should be visible
### Real behavior

I consider myself young, but even I have a tough time reading the text...
![image](https://user-images.githubusercontent.com/23620506/107130927-3cdb6780-68a0-11eb-894e-7c21bd034d73.png)

### Tested environment(s)

 * PyQtGraph version: today afternoon master branch
 * Qt Python binding: 5.14.2
 * Python version: 3.8.2

### Additional context

I work a lot with image items, so a hack I use when I am guaranteed the existence of something that can scale for me:
```python
def _paintLegend(self, painter: QtGui.QPainter, floatLegend=False):
    oldPos = self.legend.pos()
    try:
      exportScene = GraphicsScene(parent=None)
      # Let the legend grow if needed, but it can't shrink without obliterating the text
      oldScale = np.clip(0.5/np.array(self.imgItem.pixelSize()), 1, np.inf)
      # Put into a different scene so alpha doesn't render what's behind the legend in the original scene
      # Just calling legend.paint(painter) doesn't paint child items, that's why I use a different scene
      exportScene.addItem(self.legend)
      # Legend does not scale itself, handle this
      self.legend.scale(*oldScale)
      painter.save()
      if floatLegend:
        # Saves the position of the legend as it appears in the original scene,
        # compared to docking in the top-left corner
        viewPos = self.viewbox.viewRect()
        viewX = max(viewPos.x(), 0)
        viewY = max(viewPos.y(), 0)
        imPos = self.legend.mapRectToItem(self.imgItem, self.legend.rect())
        painter.translate(int(imPos.x() - viewX), int(imPos.y() - viewY))
      exportScene.render(painter)
      painter.restore()
      self.legend.scale(*(1/oldScale))
    finally:
      # Put the legend back in the original scene
      self.legend.setParentItem(self.plotItem)
      self.legend.autoAnchor(oldPos, relative=False)
```

Is there a better way of achieving the same result?

**Comment by ntjess** [comment]
Edit: After upgrading pyqtgraph, my workaround is no longer working. `scale` says there are too many arguments. Is this an intentional break from backwards compatibility?

Changing `legend.scale` to `pg.Qt.scale(legend, ...)` still works:
![image](https://user-images.githubusercontent.com/23620506/107131217-80cf6c00-68a2-11eb-88d9-10f61da14bba.png)

**Comment by j9ac9k** [comment]
@ntjess you're keeping me busy haha, thanks for uncovering; did this _ever_ work?

**Comment by ntjess** [comment]
Haha, as you probably know by know I have been building an app that relies heavily on pyqtgraph features and finally have more time to submit some bugs I've come across / features I've thought were lacking.

To my knowledge this problem has always existed, it wasn't until recently I figured out a workaround. It's a bit tough, since just calling `legend.resize()` will resize the surrounding box, but not all the legend items.

Pyqtgraph master works with `self.legend.setScale`, but will only allow fixed ratio scaling.

**Comment by j9ac9k** [comment]
I'm giving you a hard time, thanks for using the library and providing good bug reports.

I think the fix here is to inherit from `UIGraphicsItem` which is supposed to remain a fixed size and not subject itself to scaling.

**Comment by pijyoi** [comment]
>Pyqtgraph master works with self.legend.setScale, but will only allow fixed ratio scaling.

The API for scaling is:
- item.setScale(xy)
- item.setTransform(QTransform.fromScale(x, y))

**Comment by ntjess** [comment]
@j9ac9k Interesting, I don't have the graphics background to test out that solution, do you have a MWE you can point me toward? Would it be a drop-in replacement for the `GraphicsWidget` the legend currently inherits?

@pijyoi Does this mean the old `scale(x,y)` will no longer be available moving forward? I don't know if it was ever intended to be a documented feature, but it is a breaking change for anyone relying on it previously

**Comment by j9ac9k** [comment]
> @j9ac9k Interesting, I don't have the graphics background to test out that solution, do you have a MWE you can point me toward? Would it be a drop-in replacement for the GraphicsWidget the legend currently inherits?

Sorry, I meant the fix I think here is to make LegendItem inherit from `UIGraphicsItem`, not whatever you are working on.

**Comment by pijyoi** [comment]
@ntjess , it was removed in #1533 as part of general efforts to port code to support Qt6.
The 2 argument scale method was a holdover from Qt4, and it seemed odd (to me) that Qt6 should still be emulating the Qt4 API.
If you feel strongly about it, you could push for it to be restored.

**Comment by ntjess** [comment]
@pijyoi Nah, API changes are API changes. I just wanted to confirm it was an intended side effect (I know Qt4 is being phased out, I suppose it just didn't register to me this was a byproduct of the API).

**Comment by j9ac9k** [comment]
> @j9ac9k Interesting, I don't have the graphics background to test out that solution, do you have a MWE you can point me toward? Would it be a drop-in replacement for the GraphicsWidget the legend currently inherits?

I was thinking out-loud; ...as you point out, LegendItem inherits from GraphicsWidget, so UIGraphicsItem won't be a particularly good replacement...

**Comment by pijyoi** [comment]
@ntjess, does the workaround in #1767 also work for your case?

**Comment by ntjess** [comment]
@pijyoi Looks like it works for this posted example.

In my application (a bit more than the MWE) I can't call `render` directly since there are some things going on in my scene. Is there a way to programmatically add a legend to some painter without adding the legend to its own scene, calling `render` on that scene, and moving it back (the additional context)? 

I tried to steal from the SvgExporter example and paint each `item, label` in `legend.items` but it turned out quite poorly

**Comment by j9ac9k** [comment]
In case someone comes along here, the fix for this issue appears to be to remove the line

```python
        self.setFlag(self.GraphicsItemFlag.ItemIgnoresTransformations)
```

from `LegendItem.py`

**Comment by j9ac9k** [comment]
Accidental close, my mistake!
